### ISSUE TITLE
[bitnami/thanos] thanos servicemonitors are invalid in dual-stack mode

### ISSUE THREAD

**Issue by antonincms** [bitnami/charts#27103]
### Name and Version

bitnami/thanos 15.6.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Hello, 

When deploying thanos in dual-stack mode and a serviceMonitor, with the provided values the "thanos-receive" service point to the thanos-receive-distributor pods and the "thanos-receive-headless" service point to the thanos-receive pods (which I guess was done to not change the URL of data ingestion). 

Unfortunately as the "thanos-receive" serviceMonitor point to the "thanos-receive" and the  "thanos-receive-distributor" serviceMonitor does not point to any service, the thanos-receive is not scrapped and the receive-distributor scrapped by the wrong job.

### Are you using any custom parameters or values?

```yaml
receive:
  enabled: true
  mode: dual-mode # Split receiver in distributor and receiver
  service:
    additionalHeadless: true # Required for hashring to work
receiveDistributor:
  enabled: true
```

### What is the expected behavior?

The "thanos-receive" serviceMonitor should instead point to the "thanos-receive-headless" service, and the "thanos-receive-distributor" serviceMonitor should point to the "thanos-receive" service. 

### What do you see instead?

The "thanos-receive" serviceMonitor point to the "thanos-receive" service, scrapping "thanos-receive-distributor" pods, and the "thanos-receive-distributor" serviceMonitor does not point to any service.

### Additional information

I think it could be fixed by changing the labels of the "thanos-receive" service to be `app.kubernetes.io/component: receive-distributor` instead of `app.kubernetes.io/component: receive` and add `prometheus-operator/monitor: "true"` to thanos-receive-headless.

**Comment by javsalgar** [comment]
Hi!

Thank you so much for reporting the issue. Would you like to submit a PR with your proposed solution?

**Comment by antonincms** [comment]
Ok it seems the issue was functionnally fixed in #26880, but I will open my PR as I think it could be improved since :
* the servicemonitor "thanos-receive" scrape all the receive and receive-distributor pods, and "thanos-receive-distributor" scrapes nothing
* in single stack mode, both service will point to the same pods, causing them to be scrapped twice :
![Capture d’écran 2024-06-12 à 11 43 50](https://github.com/bitnami/charts/assets/9219052/2eba52a9-559b-45cc-b22f-35b6e1026e85)

**Comment by javsalgar** [comment]
Thank you so much for the PR! The team will take a look
